compiler/CMakeLists.txt

```diff
@@ -119,15 +126,50 @@ if(THEROCK_ENABLE_COMPILER)
       amd-llvm
       ${THEROCK_BUNDLED_ZSTD}
   )
-  therock_cmake_subproject_provide_package(amd-comgr amd_comgr lib/cmake/amd_comgr)
-  therock_cmake_subproject_activate(amd-comgr)
+  if(NOT THEROCK_FLAG_COMGR_DELAY_LOAD)
+    therock_cmake_subproject_provide_package("${_amd_comgr_target_name}" amd_comgr lib/cmake/amd_comgr)
+  endif()
+  therock_cmake_subproject_activate("${_amd_comgr_target_name}")
 
   therock_test_validate_shared_lib(
     PATH amd-comgr/dist/lib
     LIB_NAMES
       libamd_comgr.so
   )
 
+  ##############################################################################
+  # amd-comgr stub
+  # Generates a stub that is used for dynamically loading amd-comgr in an
+  # appropriately private namespace. Since all uses of LLVM are hidden behind
+  # the comgr interface, this allows us to completely hide our LLVM and make it
+  # impossible to conflict.
```

Comment:
How true is "Since all uses of LLVM are hidden behind the comgr interface", or what is in scope for "LLVM" there? Are binary tools like `amdclang` already self-contained / isolated enough?

Reply:
Yes, binaries we don't care about. What we are trying to protect is in-process use when ROCM is used as an API. So 100% true.